src/Raven.Server/Documents/Subscriptions/SubscriptionFetcher.cs

```diff
@@ -15,16 +17,19 @@ protected SubscriptionFetcher(DocumentDatabase database, SubscriptionConnections
             Logger = LoggingSource.Instance.GetLogger<SubscriptionFetcher<T>>(Database.Name);
         }
 
-        protected abstract IEnumerable<T> FetchByEtag();
+        protected abstract IEnumerator<T> FetchByEtag();
 
-        protected abstract IEnumerable<T> FetchFromResend();
+        protected abstract IEnumerator<T> FetchFromResend();
 
         public IEnumerable<T> GetEnumerator()
         {
             FetchingFrom = FetchingOrigin.Resend;
-            foreach (var item in FetchFromResend())
+            using (var fetchFromResend = FetchFromResend())
```

Comment:
my bad, didn't know that you can execute foreach on enumerator (since C# 9). Now we have a double dispose. One in foreach and 2nd one here.